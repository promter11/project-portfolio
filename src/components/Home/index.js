import React from "react";

import styles from "./Home.module.scss";

const Home = () => {
  return (
    <section className={styles.home}>
      <h1 className={styles.title}>Hello, I'm Alex.</h1>
      <p className={styles.desc}>
        Front-end developer who loves to work with React.
      </p>
      <a className={styles.link} href="/about">
        Learn more
      </a>
    </section>
  );
};

export default Home;
