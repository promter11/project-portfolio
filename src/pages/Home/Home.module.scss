@import "../../styles/variables";

.home {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  height: 100%;
  padding: 0 14rem;
  background: rgba(0, 0, 0, 0.75) url("../../assets/images/Home.jpg") no-repeat
    center center / cover;
  background-blend-mode: darken;
}

.container {
  max-width: 117rem;
  width: 100%;
  margin: 0 auto;
}

.title {
  max-width: 49rem;
  margin-bottom: 3rem;
  text-shadow: 0.5rem 0.5rem 0 rgba(0, 0, 0, 1);
}

.desc {
  margin-bottom: 4rem;
  font-size: 1.8rem;
  font-weight: 100;
  line-height: 150%;
  color: $white-color;
}

.link {
  display: inline-block;
  flex-shrink: 0;
  padding: 1.5rem 3rem;
  font-size: 1.4rem;
  font-weight: 100;
  line-height: 100%;
  text-transform: uppercase;
  letter-spacing: 0.02rem;
  border: 0.1rem solid $white-color;
  transition: all 0.5s ease;
  background-color: transparent;
  color: $white-color;

  &:hover {
    box-shadow: inset 7.6rem 0 0 0 $white-color,
      inset -7.6rem 0 0 0 $white-color;
    background-color: transparent;
    color: $black-color;
  }

  &:active {
    transition: none;
    transform: translateY(0.2rem);
  }
}
