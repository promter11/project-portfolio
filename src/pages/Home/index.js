import React from "react";
import { Link } from "react-router-dom";

import Container from "../../components/Container";
import Heading from "../../components/Heading";

import styles from "./Home.module.scss";

const Home = () => {
  return (
    <section className={styles.home}>
      <Container>
        <Heading className={styles.title} level={1}>
          Hello,
          <br />
          I'm Alex.
        </Heading>
        <p className={styles.desc}>
          Front-end developer who loves to work with React.
        </p>
        <Link className={styles.link} to="/about">
          Learn more
        </Link>
      </Container>
    </section>
  );
};

export default Home;
