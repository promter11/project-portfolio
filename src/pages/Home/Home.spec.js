import React from "react";
import { shallow } from "enzyme";

import Home from "./index";

const props = {
  language: "RU",
};

describe("Home page", () => {
  it("renders correctly with props", () => {
    const wrapper = shallow(<Home {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
