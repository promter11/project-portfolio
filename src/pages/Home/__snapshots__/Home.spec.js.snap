// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders correctly with props 1`] = `
<section
  className="home"
>
  <Container>
    <Heading
      className="title"
      level={1}
    >
      Привет, я Александр.
    </Heading>
    <p
      className="desc"
    >
      Фронтенд-разработчик, который любит работать с React.
    </p>
    <Link
      className="link"
      to="/about"
    >
      Узнать больше
    </Link>
  </Container>
</section>
`;
